test/widgets/test_notify.py

```diff
@@ -169,3 +170,147 @@ def test_notifications(manager_nospawn, minimal_conf_noscreen, monkeypatch):
     # Show the display
     obj.display()
     assert obj.info()["text"] == MESSAGE_3.format(colour=LOW)
+
+
+def test_capabilities():
+    # Default capabilities are "body" and "actions"
+    widget_with_actions = notify.Notify()
+    assert widget_with_actions.capabilities == {"body", "actions"}
+
+    # If the user chooses not to have actions, the capabilities
+    # are adjusted accordingly
+    widget_no_actions = notify.Notify(action=False)
+    assert widget_no_actions.capabilities == {"body"}
+
+
+@pytest.mark.skipif(
+    shutil.which("notify-send") is None,
+    reason="notify-send not installed."
+    )
+@pytest.mark.usefixtures("dbus")
+def test_invoke_and_clear(manager_nospawn, minimal_conf_noscreen):
+
+    # We need to create an object to listen for signals from the qtile
+    # notification server. This needs to be created within the manager
+    # object so we rely on "eval" applying "exec".
+    handler = textwrap.dedent(
+        """
+        import asyncio
+
+        from libqtile.utils import add_signal_receiver
+
+        class SignalListener:
+            def __init__(sl):
```

Comment:
Wasn't sure if using `self` here would be problematic given that `self` refers to the qtile instance!

Reply:
It would work fine as long as you don't need to access the qtile instance, as it would just be masked in this scope

Reply:
OK - let me change it as it would be easier to read that way.